src/dotnet/CarbonAware.Plugins.JsonReaderPlugin/CarbonAwareJsonReaderPlugin.cs

```diff
@@ -71,10 +93,13 @@ private string ReadFromResource(string key)
         return readerMetaData.ReadToEnd();
     }
  
-    protected virtual List<EmissionsData> GetSampleJson()
+    protected virtual List<EmissionsData>? GetSampleJson()
     {
-        var data = ReadFromResource("CarbonAware.Plugins.JsonReaderPlugin.test-data-azure-emissions.json");
-        var jsonObject = JsonConvert.DeserializeObject<EmissionsJsonFile>(data);
-        return jsonObject.Emissions;
+        if(emissionsData != null && !emissionsData.Any()) {
```

Comment:
@pritipath , I think this is the opposite. If the local property `emissionData == null`, fetch from the file then keep the reference.